Source/WebDriver/soup/WebSocketServerSoup.cpp

```diff
@@ -75,7 +75,7 @@ static void handleIncomingHandshake(SoupServer*, SoupServerMessage* message, con
     };
 
     auto* webSocketServer = static_cast<WebSocketServer*>(userData);
-    if (webSocketServer->messageHandler().acceptHandshake(WTFMove(handshakeMessage))) // Follow with handshake procedure
+    if (webSocketServer && webSocketServer->messageHandler()->acceptHandshake(WTFMove(handshakeMessage))) // Follow with handshake procedure
```

Comment:
webSocketServer can't be nullptr, since we are passing this to the handler as userData. We should ensure the handler can't be called when the WebSocketServer is destroyed, which should be already handled by disconnect()

Reply:
There are 3 places where we access the handlers. Two are callbacks to `soup_server_add_handler` and `soup_server_add_websocket_handler`. For these, the lifetime is already handled by the lifetime of the service in `disconnect`.

The other one is from `SoupWebsocketConnection::message` handler, which should also be disconnected automatically when we disconnect the server, right?